src/neural/blas/fully_connected_layer.h

```diff
@@ -0,0 +1,46 @@
+/*
+ This file is part of Leela Chess Zero.
+ Copyright (C) 2018 The LCZero Authors
+
+ Leela Chess is free software: you can redistribute it and/or modify
+ it under the terms of the GNU General Public License as published by
+ the Free Software Foundation, either version 3 of the License, or
+ (at your option) any later version.
+
+ Leela Chess is distributed in the hope that it will be useful,
+ but WITHOUT ANY WARRANTY; without even the implied warranty of
+ MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ GNU General Public License for more details.
+
+ You should have received a copy of the GNU General Public License
+ along with Leela Chess.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#pragma once
+
+#include <vector>
+
+namespace lczero {
+
+class FullyConnected {
+ public:
+  // From input_size to output_size (batched)
+  static void Forward(const int batch_size, const int input_size,
+                      const int output_size, const float* input,
+                      const float* weights, const float* biases,
+                      bool apply_relu, float* output);
+
+  // No batched, from input_size to scalar
+  static float ToScalar(const int input_size, const float* input,
```

Comment:
It definitely belongs in FullyConnectedLayer since it's a fully-connected layer computation (all-values to one value).

But I would like a better name. Suggest.

Reply:
`InnerProduct` was fine, imo!

Reply:
But's it's really a Matrix-Matrix multiplication (which is not an inner-product mathematically) followed by adding the bias, plus a Relu.

As I recall (and without wikipedia please) mathematically  an inner product is when you have xT M x, with M positive definite. This is an inner product.